### ISSUE TITLE
Bug: [no-deprecated] doesn't work on export/re-export statements

### ISSUE THREAD

**Issue by Hotell** [typescript-eslint/typescript-eslint#11197]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Issue Description

`no-deprecate` won't process re-exports and ignores them completely.

Our barrel file contains re-exports from sub-packages, which some are deprecated and thus need to be disabled inline ( e.g. https://github.com/microsoft/fluentui/blob/master/packages/react-components/react-components/src/index.ts#L106 )

Recently we migrated from https://github.com/gund/eslint-plugin-deprecation to official `no-deprecate` rule, but it appears the feature parity is not 100%. 

**Example 1:** 

All following APIs contain `@deprecated` pragma but nothing is being reported

```ts
// @filename react-components/src/index.ts

export {
  InfoButton,
  infoButtonClassNames,
  renderInfoButton_unstable,
  useInfoButtonStyles_unstable,
  useInfoButton_unstable,
} from '@fluentui/react-infobutton';
export type { InfoButtonProps, InfoButtonSlots, InfoButtonState } from '@fluentui/react-infobutton';
```

<img width="845" src="https://github.com/user-attachments/assets/debd60c6-a7b1-4a68-aba5-9123c0cc4ce4" alt="showcase of re-exported apis containing @deprecated jsdoc pragma"/>

**Example 2:** 

Wrong report for implicit inline `no-deprecate`

```
/Users/fluentui/packages/react-components/react-components/src/index.ts
  106:3   warning  Unused eslint-disable directive (no problems were reported from '@typescript-eslint/no-deprecated')
  111:3   warning  Unused eslint-disable directive (no problems were reported from '@typescript-eslint/no-deprecated')
  118:3   warning  Unused eslint-disable directive (no problems were reported from '@typescript-eslint/no-deprecated')
  137:3   warning  Unused eslint-disable directive (no problems were reported from '@typescript-eslint/no-deprecated')
  140:3   warning  Unused eslint-disable directive (no problems were reported from '@typescript-eslint/no-deprecated')
  244:3   warning  Unused eslint-disable directive (no problems were reported from '@typescript-eslint/no-deprecated')
  377:3   warning  Unused eslint-disable directive (no problems were reported from '@typescript-eslint/no-deprecated')
  565:3   warning  Unused eslint-disable directive (no problems were reported from '@typescript-eslint/no-deprecated')
  581:3   warning  Unused eslint-disable directive (no problems were reported from '@typescript-eslint/no-deprecated')
  633:34  warning  Unused eslint-disable directive (no problems were reported from '@typescript-eslint/no-deprecated')
```

### Reproduction Repository Link

https://github.com/microsoft/fluentui

### Repro Steps

1) clone the repo
2) `yarn install`
3) `yarn nx run react-components:lint`

**NOTE:** we use  `ESLINT_USE_FLAT_CONFIG=false` under the hood

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `8.32.0` |
| `@typescript-eslint/parser`            | `8.32.0` |
| `@typescript-eslint/scope-manager`     | `8.32.0` |
| `@typescript-eslint/typescript-estree` | `8.32.0` |
| `@typescript-eslint/type-utils`        | `8.32.0` |
| `@typescript-eslint/utils`             | `8.32.0` |
| `TypeScript`                           | `5.3.3` |
| `ESLint`                               | `9.17.0` |
| `node`                                 | `20` |

### Related issues:
- https://github.com/typescript-eslint/typescript-eslint/issues/10828
- https://github.com/typescript-eslint/typescript-eslint/issues/10327
